### DISCUSSION TITLE
Dispatch window w/o changing focus?

### DISCUSSION THREAD

**Discussion by brannegan** [hyprwm/Hyprland#739] [Q&A] [answered]
I'm trying to write [nnn](https://github.com/jarun/nnn) file manager preview plugin, but stuck upon one issue. When i select image new window is spawned, but this also change focus to newly opened window. Something like `hyprctl dispatch exec imv && hyprctl dispatch cyclenext prev` doesn't work.
Is it possible to spawn new window without changing focus?

**Comment by vaxerski** [comment, marked as answer]
no you can't. What you can do is query `hyprctl activewindow` and use that to `hyprctl dispatch focuswindow <...>`

**Reply by brannegan** [reply]
Actually that doesn't work either. Simple script to check:
```
WINDOW_ADDR=$(hyprctl activewindow -j | jq .address | tr -d \")
hyprctl dispatch exec imv 1.jpg
hyprctl dispatch focuswindow address:$WINDOW_ADDR
```
I think that because`hyprctl dispatch focuswindow` finished before `hyprctl dispatch exec`.
I don't like to stick `sleep` between `hyprctl` calls.
Any other ideas?

**Reply by vaxerski** [reply]
socket2 and wait for a focus event?

**Comment by brannegan** [comment]
It can be done. Thanks for your feedback. I think that for scripting purposes it would be nice to spawn window without loosing focus (e.g.  `hyprctl dispatch fork ...`).

**Reply by JustSimplyKyle** [reply]
`silentexec`?
Although fork is technically more "accurate" is not that easy to understand

**Comment by JustSimplyKyle** [comment]
Although, why can't you just launch stuff like the way you usually use?(`program &`)

**Reply by brannegan** [reply]
Because new window automatically gains focus.
